### PR TITLE
✨Support query NFT royalty for stakeholders

### PR DESCRIPTION
**Pull request by WeiJun0827**
- [x] Create a new table for NFT royalty
- [x] Update extractors of `MsgSend`, `buy_nft`, and `sell_nft` to write data into royalty table
- [x] Add a getter function for NFT royalty
- [x] Update tests of `TestSendNftWithPrice()`, `TestListing()`, and `TestOffer()`
- [x] Add commands for migrating existing txs
- [x] Add an endpoint for querying NFT royalty
- [x] Add a `total_royalties` filed in `/user-stat` API

### CODE REVIEW

File: db/extract.go [hunk 1]
```diff
@@ -225,12 +225,16 @@ func (batch *Batch) InsertNft(n Nft) {
...
+		metadata, latest_price, price_updated_at
+	)
 	VALUES
-	($1, $2, $3, $4, $5, $6)
+	($1, $2, $3, $4, $5, $6, $7, $8)
 	ON CONFLICT DO NOTHING`
-	batch.Batch.Queue(sql, n.NftId, n.ClassId, n.Owner, n.Uri, n.UriHash, n.Metadata)
+	batch.Batch.Queue(sql,
+		n.NftId, n.ClassId, n.Owner, n.Uri, n.UriHash,
+		n.Metadata, n.LatestPrice, n.Timestamp)
```
Comment (nnkken) [inline db/extract.go:237]: 1. when a NFT is minted, we may not want to set `LatestPrice` to 0 (the default value in Go struct for `uint64`) but instead leave it as `null` in the table
2. `Timestamp` and `PriceUpdatedAt` are different fields

File: db/schema/parallel/nft_royalty.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+				}
+
+				events, err := utils.ParseEvents(eventRaw)
+				if err != nil {
+					logger.L.Errorw("Error when parsing events", "error", err)
+					return err
+				}
+
+				msgAction := utils.GetEventsValue(events, "message", "action")
+				if msgAction == "/cosmos.bank.v1beta1.MsgSend" || msgAction == string(db.ACTION_BUY) || msgAction == string(db.ACTION_SELL) {
```
Comment (nnkken) [inline db/schema/parallel/nft_royalty.go:67]: Should use `db.ACTION_SEND`

File: db/schema/parallel/nft_royalty.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+				_, err = dbTx.Exec(context.Background(), `
+						INSERT INTO nft_royalty (class_id, nft_id, tx_hash, stakeholder_address, royalty)
+						VALUES ($1, $2, $3, $4, $5);
+					`, royalty.ClassId, royalty.NftId, royalty.TxHash, royalty.Stakeholder, royalty.Royalty)
+				if err != nil {
+					logger.L.Errorw("Error when inserting into nft_royalty", "error", err)
+					return err
+				}
+			}
+			lastRoyalty := royalties[count-1]
```
Comment (nnkken) [inline db/schema/parallel/nft_royalty.go:92]: What if `count == 0`?

File: test/test_cleanup_data.sql [hunk 4]
```diff
@@ -6,6 +6,7 @@ DELETE FROM nft_event;
 DELETE FROM nft;
 DELETE FROM nft_class;
 DELETE FROM nft_marketplace;
+DELETE FROM nft_royalty;
```
Comment (nnkken) [inline test/test_cleanup_data.sql:9]: Please also add cleanup in `test_cleanup_table.sql`

File: db/schema/parallel/nft_royalty.go (hunk 3 above)
Comment (WeiJun0827) [reply db/schema/parallel/nft_royalty.go:92]: I should handle it, thanks!

File: db/schema/parallel/nft_royalty.go (hunk 2 above)
Comment (WeiJun0827) [reply db/schema/parallel/nft_royalty.go:67]: Does `db.ACTION_SEND` refer to `/cosmos.nft.v1beta1.MsgSend`? 
I think the events of `ACTION_SEND` in `logs` array of a tx ([sample](https://mainnet-node.like.co/cosmos/tx/v1beta1/txs/DF3D66EC7C9EAE28EE076D4FC80268FA42595957BC323B9C48A879EB453BDA22)) is as the below format:
```
{
  "msg_index": 1,
  "log": "",
  "events": [
    {
      "type": "cosmos.nft.v1beta1.EventSend",
      "attributes": [
        {
          "key": "sender",
          "value": "\"like17m4vwrnhjmd20uu7tst7nv0kap6ee7js69jfrs\""
        },
        {
          "key": "receiver",
          "value": "\"like1a33400g7wrlv04tzdjzqwpar5rs4qhkakz8y3w\""
        },
        {
          "key": "class_id",
          "value": "\"likenft1t5ftaxzyrnafewsav8qxvvsafs59a8xe2wnrl594yag04u9zwsjqrq0vgt\""
        },
        {
          "key": "id",
          "value": "\"writing-1a4bc1b2-fe70-4033-9c76-1424b66d4f16\""
        }
      ]
    },
    {
      "type": "message",
      "attributes": [
        {
          "key": "action",
          "value": "/cosmos.nft.v1beta1.MsgSend"
        }
      ]
    }
  ]
},
```
Should we instead fetch the `coin_received` event attributes from `/cosmos.bank.v1beta1.MsgSend`?

Comment (nnkken) [reply db/schema/parallel/nft_royalty.go:67]: Oh I missed the module name, I thought it was `/cosmos.nft.v1beta1.MsgSend`

**Comment by WeiJun0827** [comment]
I reset my branch to the wrong commit, the PR was accidentally closed...

**Pull request closed without merging**